### PR TITLE
juju 2.8.0

### PR DESCRIPTION
**Pull request by hpidcock**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Thank-you for looking at this PR.
The PR bumps juju to 2.8.0
It also introduces a few build flags that juju uses now for extra version information.

### CODE REVIEW

File: Formula/juju.rb [hunk 1]
```diff
@@ -15,8 +15,11 @@ class Juju < Formula
 
   def install
     ENV["GOPATH"] = buildpath
-    system "go", "build", "github.com/juju/juju/cmd/juju"
-    system "go", "build", "github.com/juju/juju/cmd/plugins/juju-metadata"
+    git_commit = "d816abe62fbf6787974e5c4e140818ca08586e44"
```
Comment (dawidd6) [inline Formula/juju.rb:18]: Can we somehow determine this hash automatically via `git`?

Comment (hpidcock) [reply Formula/juju.rb:18]: @dawidd6 not without introducing a git resource, which would include the commit sha anyway. The reason we use the launchpad tar is it includes go vendored dependencies.

In our next release we will include a file that contains the git commit, at that point we can replace the hardcoded git_commit here with that file's contents.

Comment (SMillerDev) [reply Formula/juju.rb:18]: I'd just change the download strategy to download tagged git releases. It's way easier.

File: Formula/juju.rb [hunk 2]
```diff
@@ -14,11 +15,15 @@ class Juju < Formula
   depends_on "go" => :build
 
   def install
-    ENV["GOPATH"] = buildpath
-    system "go", "build", "github.com/juju/juju/cmd/juju"
-    system "go", "build", "github.com/juju/juju/cmd/plugins/juju-metadata"
-    bin.install "juju", "juju-metadata"
-    bash_completion.install "src/github.com/juju/juju/etc/bash_completion.d/juju"
+    git_commit = Utils.popen_read("git", "rev-parse", "HEAD").chomp
```
Comment (SMillerDev) [inline Formula/juju.rb:18]:
```suggestion
    git_commit = Utils.safe_popen_read("git", "rev-parse", "HEAD").chomp
```

File: Formula/juju.rb [hunk 3]
```diff
@@ -14,11 +15,15 @@ class Juju < Formula
...
+    ld_flags = %W[
+      -s -w
+      -X version.GitCommit=#{git_commit}
+      -X version.GitTreeState=clean
+    ]
+    system "go", "build", "-ldflags", ld_flags.join(" "), "-o", bin/"juju", "./cmd/juju"
+    system "go", "build", "-ldflags", ld_flags.join(" "), "-o", bin/"juju-metadata", "./cmd/plugins/juju-metadata"
```
Comment (SMillerDev) [inline Formula/juju.rb:25]: Can this use `*std_go_args`

**Review by dawidd6 — APPROVED** [review]

**Pull request closed without merging**